### DISCUSSION TITLE
Haskell code blocks and examples with output

### DISCUSSION THREAD

**Discussion by lambdacalculator** [quarto-dev/quarto-cli#3812] [Q&A]
Have executable Haskell code blocks been implemented anywhere? In general, what approach do writers of Haskell tutorials/books take to include code examples with automatically generated output (say from ghci)? What if library imports are needed to produce the output?

**Comment by mcanouil** [comment]
The `knitr` engine supports Haskell.

````markdown
---
engine: knitr
---

```{haskell}
code
```
````
